File: src/main/java/me/kikugie/techutils/TechUtilsMod.java
package me.kikugie.techutils;

import fi.dy.masa.malilib.event.InitializationHandler;
import me.kikugie.techutils.command.IsorenderSelectionCommand;
import me.kikugie.techutils.feature.worldedit.WorldEditActionManager;
import me.kikugie.techutils.networking.GamerQueryHandler;
import me.kikugie.techutils.utils.ResponseMuffler;
import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback;
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class TechUtilsMod implements ClientModInitializer {
    public static final Logger LOGGER = LoggerFactory.getLogger(Reference.MOD_ID);

    @Override
    public void onInitializeClient() {
        InitializationHandler.getInstance().registerInitializationHandler(new InitHandler());
        ClientCommandRegistrationCallback.EVENT.register(IsorenderSelectionCommand::register);
        ClientPlayConnectionEvents.JOIN.register((handler, sender, client) -> WorldEditActionManager.initManager());
        ClientPlayConnectionEvents.DISCONNECT.register((handler, client) -> {
            WorldEditActionManager.reset();
            ResponseMuffler.clear();
        });
        ClientTickEvents.START_CLIENT_TICK.register((client) -> {
            GamerQueryHandler.INSTANCE.onTick();
        });
    }
}

File: src/main/kotlin/me/kikugie/techutils/TechUtilsMod.kt
package me.kikugie.techutils

import fi.dy.masa.malilib.event.InitializationHandler
import me.kikugie.techutils.feature.WorldEditActionManager
import me.kikugie.techutils.utils.ResponseMuffler
import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents
import net.fabricmc.fabric.api.networking.v1.PacketSender
import net.minecraft.client.MinecraftClient
import net.minecraft.client.network.ClientPlayNetworkHandler
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class TechUtilsMod : ModInitializer {
    override fun onInitialize() {
        InitializationHandler.getInstance().registerInitializationHandler(InitHandler())
        ClientPlayConnectionEvents.JOIN.register(ClientPlayConnectionEvents.Join { _: ClientPlayNetworkHandler?, _: PacketSender?, _: MinecraftClient? -> WorldEditActionManager.init() })
        ClientPlayConnectionEvents.DISCONNECT.register(ClientPlayConnectionEvents.Disconnect { _: ClientPlayNetworkHandler?, _: MinecraftClient? ->
            WorldEditActionManager.reset()
            ResponseMuffler.clear()
        })
    }

    companion object {
        @JvmField
        val LOGGER: Logger = LoggerFactory.getLogger("tech-utils")
    }
}